src/resolvers/snapshot.ts

```diff
@@ -4,17 +4,20 @@ import { max } from '../constants.json';
 import { fetchHttpImage, axiosDefaultParams } from './utils';
 
 const HUB_URL = process.env.HUB_URL ?? 'https://hub.snapshot.org';
+const HUB_API_KEY = process.env.HUB_API_KEY ?? undefined;
 
 function createPropertyResolver(property: 'avatar' | 'cover') {
   return async (address: string) => {
     try {
+      const headers: Record<string, string> = HUB_API_KEY ? { 'x-api-key': HUB_API_KEY } : {};
       const user = (
         await axios({
           url: `${HUB_URL}/graphql`,
           method: 'post',
           data: {
             query: `query { user(id: "${address}") { ${property} } }`
           },
+          headers,
```

Comment:
We add a filter to this `header` here, to filter out all keys with `undefined` as value.

So we don't have to check if key is valid or not elsewhere